src/python/pants/backend/javascript/nodejs_project_test.py

```diff
@@ -71,6 +71,45 @@ def test_parses_projects(rule_runner: RuleRunner) -> None:
     assert {project.root_dir for project in projects} == {"src/js/foo", "src/js/bar"}
 
 
+def test_immutable_install_args_property(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "src/js/foo/BUILD": "package_json()",
+            "src/js/foo/package.json": given_package("foo", "0.0.1"),
+            "src/js/bar/BUILD": "package_json()",
+            "src/js/bar/package.json": given_package(
+                "bar", "0.0.2", package_manager="yarn@1.22.19"
+            ),
+            "src/js/baz/BUILD": "package_json()",
+            "src/js/baz/package.json": given_package("baz", "0.0.3", package_manager="yarn@2.4.3"),
+            "src/js/qux/BUILD": "package_json()",
+            "src/js/qux/package.json": given_package("qux", "0.0.4", package_manager="pnpm@7.5.0"),
+        }
+    )
+    projects = rule_runner.request(AllNodeJSProjects, [])
+    assert {project.immutable_install_args for project in projects} == {
```

Comment:
Thanks for writing some tests for this. For this particular test, I think it'd be good to match each particular package manager to the expected args, a bit different to the other tests in this file.

In particular, because this is a set, I think the old code would pass this this test, as the LHS and RHS will be `{("clean-install,), ("install", "--frozen-lockfile"), ("install", "--immutable")}` in for both old and new (just the counts of will change, but that's not captured).

A different way to phrase it might be by parametrizing for each of the package mangers:

```python
@pytest.mark.parametrize(
    ("package_manager", "expected_args"),
    [ 
        (None, ("clean-install",)),
        ("yarn@1.22.19", ("install", "--frozen-lockfile"))
        ...
    ]
)
def test_immutable_install_args_property(package_manager: None | str, expected_args: tuple[str, ...], rule_runner: RuleRunner) -> None:
   # construct one package and check its args specifically
```

What do you think?

Reply:
Thats an awesome suggestion, thanks.
Changed to use parametrize: https://github.com/pantsbuild/pants/pull/20540/commits/5177a9bad79bc4813b68aafabc5e38edee5e0b91